File: ISAProject/src/main/java/com/app/grade/GradeService.java
package com.app.grade;

import java.util.List;

import com.app.guest.Guest;
import com.app.restaurant.Restaurant;

public interface GradeService {

	Iterable<Grade> findAll();
	Grade findOne(Long id);
	Grade save(Grade grade);
	void delete(Long id);
	Iterable<Grade> findByGuest(String email);
	List<Grade> findByRestaurant(Long id);
	List<RestaurantDTO> getRestaurantsDTO(Iterable<Restaurant> allRestaurants,Guest guest);
	GradeDTO findByWaiter(String email);
	GradeDTO findByFood(Long id);
	GradeDTO findRestaurantGrade(Long id);
}

File: ISAProject/src/main/webapp/grade/gradeService.js
app.service('gradeService', ['$http', function($http){


    var url='/grades';
	
	this.findAll = function(){
		return $http.get(url);
	}
    this.findOne = function(id){
		return $http.get(url,id);
	}  
    this.update = function(user){
		return $http.put(url, user);
	}
    this.save = function(user, restaurant){
		return $http.post(url, user, restaurant);
	}
    this.findByWaiter = function(email){
		return $http.get(url+"/findGradeForWaiter?email="+email);
	}
    this.findByRestaurant = function(id){
		return $http.get(url+"/findGradeForRestaurant/"+id);
	}
    this.findByFood = function(id){
		return $http.get(url+"/findGradeForFood/"+id);
	}

}]);